### PR TITLE
#161959254 update comments

### PR DESCRIPTION
**Pull request by MuhanguziDavid**
### What does this PR do?

It adds functionality to edit comments

### Description of tasks to be completed?

Currently, a user cannot edit their comments.
This PR adds functionality to allow users to edit their comments

### How should this be manually tested?

- Run `npm install` to install all dependencies.
- Run `npm start`
- Click the comment icon on an article.
- Click the edit icon on a comment.
- Update the comment in the modal the pops up.
- Click the submit button to save edited changes.

### What are the relevant pivotal tracker stories?

[#161959254](https://www.pivotaltracker.com/story/show/161959254)

### Any background context you want to add?

N/A

## Checklist: I completed these to help reviewers

- [x] My IDE is configured to follow the [**ESlint Conventions**](https://github.com/eslint/eslint)

- [x] I have **added tests** to cover my changes.

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the develop branch.

### CODE REVIEW

File: src/tests/actions/articleActions.test.js [hunk 1]
```diff
@@ -178,6 +182,63 @@ describe('articleActions', () => {
...
+    await flushAllPromises();
+    expect(store.getActions()).toEqual(
+      [
+        { type: UPDATE_COMMENT_INITIATED, payload: true },
+        { type: UPDATE_COMMENT_SUCCESS, payload: true },
+      ],
+    );
+  });
+
+  it('should not update a comment when a user is logged in', async () => {
```
Comment (samuelbwambale) [inline src/tests/actions/articleActions.test.js]: @MuhanguziDavid Please cross-check this comment. I think it should be `when a user is not logged in`

Comment (MuhanguziDavid) [reply src/tests/actions/articleActions.test.js]: Thank you @samuelbwambale , naming has been changed accordingly.

**Review by samuelbwambale — APPROVED** [review]
LGTM

File: src/tests/components/UpdateComment.test.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    wrapper.setProps({ ...nextProps });
+    expect(window.location.reload).toBeCalled();
+  });
+
+  it('should call editComment when handleSubmit is called', () => {
+    wrapper.instance().handleSubmit(getEvent());
+    expect(editComment).toBeCalled();
+  });
+
+  it('should set state when toggle is called is called', () => {
```
Comment (mendozabree) [inline src/tests/components/UpdateComment.test.js]: line 89 and 94, remove the extra `is called`

File: src/components/comments/CommentList.js [hunk 3]
```diff
@@ -2,20 +2,22 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import RetrieveCommentForm from './RetrieveCommentForm';
 
-export const CommentList = ({ comments }) => (
+export const CommentList = ({ comments, article }) => (
   <div>
     {comments.map(comment => (
-      <RetrieveCommentForm key={comment.id} comment={comment} />
+      <RetrieveCommentForm key={comment.id} comment={comment} article={article} />
```
Comment (RonKbS) [inline src/components/comments/CommentList.js]: Nice work @MuhanguziDavid,

Is it not possible to pass the comment alone and retrieve the id from it in the `RetrieveCommentForm` component?
That way there is no need to pass in a key that already exists in the comment

File: src/tests/components/UpdateComment.test.js (hunk 2 above)
Comment (MuhanguziDavid) [reply src/tests/components/UpdateComment.test.js]: Thanks, @mendozabree , the repeated words have been removed.

File: src/components/comments/CommentList.js (hunk 3 above)
Comment (MuhanguziDavid) [reply src/components/comments/CommentList.js]: Yes, @RonKbS , it is possible. Changes have been made accordingly.

**Review by RonKbS — APPROVED** [review]
LGTM

**Review by mendozabree — APPROVED** [review]
LGTM

**Pull request merged**